operator/pkg/declarative/reconciler.go

```diff
@@ -0,0 +1,405 @@
+package declarative
+
+import (
+	"context"
+	"fmt"
+	"github.com/go-logr/logr"
+	"github.com/kyma-project/manifest-operator/operator/pkg/custom"
+	"github.com/kyma-project/manifest-operator/operator/pkg/manifest"
+	"github.com/kyma-project/manifest-operator/operator/pkg/types"
+	"helm.sh/helm/v3/pkg/cli"
+	"helm.sh/helm/v3/pkg/strvals"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/record"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"strings"
+)
+
+var _ reconcile.Reconciler = &ManifestReconciler{}
+
+const deletionFinalizer = "custom-deletion-finalizer"
+
+type BaseCustomObject interface {
+	runtime.Object
+	metav1.Object
+}
+
+type ManifestReconciler struct {
+	prototype    BaseCustomObject
+	nativeClient client.Client
+	config       *rest.Config
+
+	mgr manager.Manager
+
+	// recorder is the EventRecorder for creating k8s events
+	recorder record.EventRecorder
```

Comment:
If you have the manager you dont need to inject a Recorder

Reply:
this is not injected, but is only an internal struct variable of `declarative.ManifestReconciler`

Reply:
I am saying that if you have the manager, you dont need to instantiate the recorder because you can create one yourself. I think this is not necessary to pass as a variable since you also pass the manager